src/analyzer/Analyzer.cc

```diff
@@ -801,4 +801,30 @@ TEST_SUITE("Analyzer management") {
         CHECK(conn->FindAnalyzer("IMAP"));
         conn->Done();
     }
+
+    TEST_CASE("Analyzer mapping") {
+        REQUIRE(zeek::analyzer_mgr);
+
+        zeek::Packet p;
+        zeek::ConnTuple t;
+        auto conn = std::make_unique<zeek::Connection>(zeek::detail::ConnKey(t), 0, &t, 0, &p);
+
+        auto ssh = zeek::analyzer_mgr->InstantiateAnalyzer("SSH", conn.get());
+        REQUIRE(ssh);
+        auto imap = zeek::analyzer_mgr->InstantiateAnalyzer("IMAP", conn.get());
+        REQUIRE(imap);
+
+        zeek::analyzer_mgr->AddComponentMapping(ssh->GetAnalyzerTag(), imap->GetAnalyzerTag());
+        zeek::analyzer_mgr->DisableAnalyzer(ssh->GetAnalyzerTag()); // needs to be disabled for mapping to take effect
+        auto ssh_is_imap = zeek::analyzer_mgr->InstantiateAnalyzer("SSH", conn.get());
+        CHECK_EQ(ssh_is_imap->GetAnalyzerTag(), imap->GetAnalyzerTag()); // SSH is now IMAP
+
+        // orderly cleanup through connection
+        auto* tcp = new zeek::packet_analysis::TCP::TCPSessionAdapter(conn.get());
+        conn->SetSessionAdapter(tcp, nullptr);
+        tcp->AddChildAnalyzer(ssh);
+        tcp->AddChildAnalyzer(imap);
+        tcp->AddChildAnalyzer(ssh_is_imap);
+        conn->Done();
```

Comment:
> None of this will run if any of the above REQUIRE fails.

That cleanup logic is mainly there to avoid the leak checker triggering on the test. If the test fails for another reason (like the `REQUIRE`), the leak doesn't really matter.

>  were great if testing with zeek::Connection would be easier and safer 

Agree, it would be great. :-) But that feels out of scope here, it's an existing issue, the new test just follows the existing pattern.

Reply:
FTR, I ran the test with the cleanup code removed and it failed even without LSAN. Fine either way though.

Reply:
I ran `zeek --test` which gave me an `abort()` triggered by this not very RAII-friendly code (to be fair, and it flagged exactly what was the problem, i.e., that `Conn::Done` was not called 🤷):

https://github.com/zeek/zeek/blob/966d411ed52ff23a48c9eca88ef7aca9068b1b02/src/Conn.cc#L71-L72